src/PostgREST/Request/ApiRequest.hs

```diff
@@ -494,7 +494,21 @@ findProc qi argumentsKeys paramsAsSingleObject allProcs contentType isInvPost =
     [proc] -> Right proc
     procs  -> Left $ AmbiguousRpc (toList procs)
   where
-    matchProc = filter matchesParams $ M.lookupDefault mempty qi allProcs -- first find the proc by name
+    matchProc =
+      if null (fst findProcPartition) && isInvPost && contentType == CTApplicationJSON
+        then snd findProcPartition -- use the fallback function if none is found with the provided parameters
+        else fst findProcPartition
```

Comment:
I think we can move this part into the `case matchProc of` above. I feel like that would make the logic a bit clearer. Something like:

```haskell
  case findProcPartition of
    ([], [])     -> Left $ NoRpc (qiSchema qi) (qiName qi) (S.toList argumentsKeys) paramsAsSingleObject contentType isInvPost
    -- no matches with named arguments, fallback to single unnamed argument
    ([], [proc]) -> Right proc
    ([], procs)  -> Left $ AmbiguousRpc (toList procs)
    -- matches with named arguments
    ([proc], _)  -> Right proc
    (procs, _)   -> Left $ AmbiguousRpc (toList procs)
```